homeassistant/components/drop_connect/coordinator.py

```diff
@@ -0,0 +1,66 @@
+"""DROP device data update coordinator object."""
+from __future__ import annotations
+
+from collections.abc import Callable
+import logging
+from typing import TYPE_CHECKING
+
+import attr
+from dropmqttapi.mqttapi import DropAPI
+
+from homeassistant.components import mqtt
+from homeassistant.components.mqtt import ReceiveMessage
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import CONF_DATA_TOPIC, CONF_DEVICE_TYPE, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class DROPDeviceDataUpdateCoordinator(DataUpdateCoordinator):
+    """DROP device object."""
+
+    _drop_api: DropAPI | None = None
+    _unsubscribe_callback: Callable[[], None] | None = attr.ib()
+
+    def __init__(self, hass: HomeAssistant, unique_id: str) -> None:
+        """Initialize the device."""
+        super().__init__(hass, _LOGGER, name=f"{DOMAIN}-{unique_id}")
+        if TYPE_CHECKING:
+            assert self.config_entry is not None
```

Comment:
```suggestion
    config_entry: ConfigEntry

    def __init__(self, hass: HomeAssistant, unique_id: str) -> None:
        """Initialize the device."""
        super().__init__(hass, _LOGGER, name=f"{DOMAIN}-{unique_id}")
```

Reply:
@pfrazer Please commit this too and remove the asserts an checks as suggested

Reply:
Changed in ff80af4b820b7e89659c2a597bdc86ab1f28fd5d.